kedro-datasets/kedro_datasets/dask/csv_dataset.py

```diff
@@ -0,0 +1,125 @@
+"""``CSVDataset`` is a data set used to load and save data to CSV files using Dask
+dataframe"""
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import Any
+
+import dask.dataframe as dd
+import fsspec
+from kedro.io.core import AbstractDataset, get_protocol_and_path
+
+
+class CSVDataset(AbstractDataset[dd.DataFrame, dd.DataFrame]):
```

Comment:
I think I'd prefer "DaskCSVDataset" to make extra clear and avoid confusion with pandas one, but the catalog would refer to it as "dask.CSVDataset" which is already explicit, and the parquet dataset does not have this "Dask" prefix, so I am fine with letting it as is.

Reply:
I think there's a precedent for this naming - eg `pandas.ParquetDataset` and `dask.ParquetDataset`, `pandas.JSONDataset`, `json.JSONDataset` and `tracking.JSONDataset`

Reply:
Let's keep it consistent then 👍